SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+#include "SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.h"
+#include "Tests/AutomationEditorCommon.h"
+
+DEFINE_LOG_CATEGORY(LogGenerateTestMapsCommandlet);
+
+UGenerateTestMapsCommandlet::UGenerateTestMapsCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+
+	ConstructorHelpers::FObjectFinder<UStaticMesh> PlaneAsset(TEXT("StaticMesh'/Engine/BasicShapes/Plane.Plane'"));
+	check(PlaneAsset.Succeeded());
+	PlaneStaticMesh = PlaneAsset.Object;
+
+	ConstructorHelpers::FObjectFinder<UMaterial> MaterialAsset(TEXT("Material'/Engine/BasicShapes/BasicShapeMaterial.BasicShapeMaterial'"));
+	check(MaterialAsset.Succeeded());
+	BasicShapeMaterial = MaterialAsset.Object;
+}
+
+UWorld* UGenerateTestMapsCommandlet::CreateNewTestMapWorld()
+{
+	UWorld* CurrentWorld = FAutomationEditorCommonUtils::CreateNewMap();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	AStaticMeshActor* Plane =
+		CastChecked<AStaticMeshActor>(GEditor->AddActor(CurrentLevel, AStaticMeshActor::StaticClass(), FTransform::Identity));
+	Plane->GetStaticMeshComponent()->SetStaticMesh(PlaneStaticMesh);
+	Plane->GetStaticMeshComponent()->SetMaterial(0, BasicShapeMaterial);
+	Plane->SetActorScale3D(FVector(
+		500, 500, 1)); // Make the initial platform larger so things don't fall off for tests which use a large area (visibility test)
+
+	AExponentialHeightFog* Fog =
+		CastChecked<AExponentialHeightFog>(GEditor->AddActor(CurrentLevel, AExponentialHeightFog::StaticClass(), FTransform::Identity));
+
+	APlayerStart* PlayerStart =
+		CastChecked<APlayerStart>(GEditor->AddActor(CurrentLevel, APlayerStart::StaticClass(), FTransform(FVector(-500, 0, 100))));
+
+	ASkyLight* SkyLight = CastChecked<ASkyLight>(GEditor->AddActor(CurrentLevel, ASkyLight::StaticClass(), FTransform::Identity));
+
+	// Use current settings for all these generated maps as default.
+	// This will probably change with UNR-4801
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->TestingSettings.TestingMode = EMapTestingMode::UseCurrentSettings;
+	WorldSettings->bEnableDebugInterface = true; // Setting to true for all test maps by default, TODO: consult Nic
+
+	// TODO: Maybe figure out how to create folders to put the actors into (just visual if someone opens the map in the editor)
+	// TODO: Maybe figure out how to set the default viewpoint when opening the map, so we don't start in the ground
+
+	return CurrentWorld;
+}
+
+// This map is mostly for native-spatial compatibility tests, so things that work in native that should also work under our networking
+// If you can, please add your test to this map, the more tests we have in a map, the more efficient it is to run them (no need for a new
+// session / deployment for every test). That does mean that tests here should ideally create everything they need dynamically, and clean up
```

Comment:
How expensive is it for a new deployment / map / session per test?

Reply:
@Vix2021 this touches your work - will depend on how expensive it is to do it now with the runtime changes.